### PR TITLE
Refactor: libcrmcluster: pcmk__node_status_t:id -> cluster_layer_id

### PR DESCRIPTION
**Pull request by nrwahl2**
This is all I'll be able to get in before EOD, so here ya go.

### CODE REVIEW

File: include/crm/cluster/internal.h [hunk 1]
```diff
@@ -93,6 +89,12 @@ typedef struct pcmk__node_status {
 
     time_t when_member;         // Since when node has been a cluster member
     time_t when_online;         // Since when peer has been online in CPG
+
+    /* @TODO The following are currently needed only by the Corosync stack.
```
Comment (nrwahl2) [inline include/crm/cluster/internal.h:93]: Turns out `cluster_layer_id` is used in the controller and attribute manager too

File: lib/cluster/membership.c [hunk 2]
```diff
@@ -728,10 +731,10 @@ search_cluster_member_cache(unsigned int id, const char *uname,
     } else if(by_id == NULL && by_name) {
         crm_trace("Only one: %p for %u/%s", by_name, id, uname);
 
-        if(id && by_name->id) {
+        if (id && by_name->cluster_layer_id) {
```
Comment (nrwahl2) [inline lib/cluster/membership.c:734]: compare to NULL, but I can get this in the next PR if the rest is okay

File: include/crm/cluster/internal.h (hunk 1 above)
Comment (kgaillot) [reply include/crm/cluster/internal.h:93]: xml_id would probably be more appropriate for those uses (they're the same for Corosync nodes, just string vs int, so it wouldn't matter currently)

**Review by kgaillot — APPROVED** [review]

File: include/crm/cluster/internal.h (hunk 1 above)
Comment (nrwahl2) [reply include/crm/cluster/internal.h:93]: agreed, slightly less efficient but not significant

**Pull request merged**